go.mod

```diff
@@ -38,6 +38,7 @@ require (
 	github.com/cosmos/gogoproto v1.4.11
 	github.com/cosmos/ibc-go/modules/capability v1.0.0
 	github.com/cosmos/ibc-go/v8 v8.0.0
+	github.com/cosmos/interchain-security/v3 v3.2.0-consumer-rc0.0.20231123140529-1819e73f6197
```

Comment:
Additionally, is there a way not to import this in Ignite? This will block in v29/v30 release on the slowest dependency to update, in this case ics. Where we won't be able to update ignite to v0.51 or ibc v9 before this gets released

Reply:
> Additionally, is there a way not to import this in Ignite?

Maybe, the only place where this dependency is actually used is for the consumer genesis module (see `ignite/services/chain/init.go#WriteConsumerGenesis` function).

Reply:
I'm not sure if the consumer module can start without a valid genesis, which is problematic for the `ignite chain serve` command.

The reason I'm not sure is that I can't test it, the current version fails before that point due to a new depinject recently introduced in `root.go`. 

```
✘ Panic: can't resolve type github.com/cosmos/cosmos-sdk/x/slashing/types/types.StakingKeeper for github.com/cosmos/cosmos-sdk/x/slashing.ProvideModule (/home/tom/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.50.2/x/slashing/module.go:230):
	while resolving:                                                                                                                                                                                                                           
		types.StakingKeeper for github.com/cosmos/cosmos-sdk/x/slashing.ProvideModule (/home/tom/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.50.2/x/slashing/module.go:230)                                                                         
		store.KVStoreService for github.com/cosmos/cosmos-sdk/x/slashing.ProvideModule (/home/tom/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.50.2/x/slashing/module.go:230)                                                                        
		map[string]appmodule.AppModule for testchainconsumer/cmd/testchainconsumerd/cmd.NewRootCmd (/home/tom/src/ignite/testchainconsumer/cmd/testchainconsumerd/cmd/root.go:41)                                                                  
		ante.FeegrantKeeper for github.com/cosmos/cosmos-sdk/x/auth/tx/config.ProvideModule (/home/tom/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.50.2/x/auth/tx/config/config.go:68)                                                              
		depinject.ModuleKey for github.com/cosmos/cosmos-sdk/x/params.ProvideSubspace (/home/tom/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.50.2/x/params/module.go:146)                                                                           
		exported.Subspace for github.com/cosmos/cosmos-sdk/x/auth.ProvideModule (/home/tom/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.50.2/x/auth/module.go:209)                                                                                   
		func() types.AccountI for github.com/cosmos/cosmos-sdk/x/auth.ProvideModule (/home/tom/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.50.2/x/auth/module.go:209)                                                                               
		types.RandomGenesisAccountsFn for github.com/cosmos/cosmos-sdk/x/auth.ProvideModule (/home/tom/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.50.2/x/auth/module.go:209)                                                                       
		store.KVStoreService for github.com/cosmos/cosmos-sdk/x/auth.ProvideModule (/home/tom/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.50.2/x/auth/module.go:209)                                                                                
		types.AccountKeeper for github.com/cosmos/cosmos-sdk/x/bank.ProvideModule (/home/tom/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.50.2/x/bank/module.go:227)                                                                                 
		store.KVStoreService for github.com/cosmos/cosmos-sdk/x/bank.ProvideModule (/home/tom/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.50.2/x/bank/module.go:227)                                                                                
		types.BankKeeper for github.com/cosmos/cosmos-sdk/x/auth/tx/config.ProvideModule (/home/tom/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.50.2/x/auth/tx/config/config.go:68)                                                                 
		func() runtime.ValidatorAddressCodec for github.com/cosmos/cosmos-sdk/runtime.ProvideAddressCodec (/home/tom/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.50.2/runtime/module.go:281)                                                        
		func() address.Codec for github.com/cosmos/cosmos-sdk/runtime.ProvideAddressCodec (/home/tom/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.50.2/runtime/module.go:281)                                                                        
		*modulev1.Module for github.com/cosmos/cosmos-sdk/runtime.ProvideAddressCodec (/home/tom/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.50.2/runtime/module.go:281)                                                                            
		address.Codec for github.com/cosmos/cosmos-sdk/x/auth/tx/config.ProvideModule (/home/tom/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.50.2/x/auth/tx/config/config.go:68)                                                                    
		tx.ConfigOptions for testchainconsumer/cmd/testchainconsumerd/cmd.NewRootCmd (/home/tom/src/ignite/testchainconsumer/cmd/testchainconsumerd/cmd/root.go:41)         
```

Reply:
I know @Ehsan-saradar mentioned about an Ignite app. That would solve the depending issue as the feature would be managed by an app. And if we want to migrate to the latest version we won't have to wait ics support.

Concerning depinject, happy to have a look later today.

Reply:
Ignite apps (formerly plugins, I still prefer this name, I found apps confusing since we already have this `App` struct for the blockchain application) are just external programs. An external program cannot update the chain scaffolding, as we need to do to create a consumer chain, without duplicating most of the code and most of the templates. I don't see a consumer chain scaffolding plugin as a viable solution. It would just make maintenance a PITA because of all the duplication.

But maybe you can change my mind :)

Reply:
Maybe this is a good point to start introducing new endpoints to the plugin's ClientAPI with the purpose of allowing scaffolding. @tbruyelle we could explore that functionality if it makes sense and solves this issue.

Reply:
I'm not sure how the ClientAPI could help us on this, see my answer [here](https://github.com/ignite/cli/pull/3899#issuecomment-1903710239).

Reply:
Also one idea that came to my mind and i wanted to share with you is that if it's possible to use hooks on `ignite s chain` instead of using `Execute` function?

Reply:
Hey @Ehsan-saradar , thanks for your suggestion, unfortunately, hooks cannot be used for consumer chain.

As mentioned elsewhere, the ICS dependency is required for 2 tasks:
1. check if consumer chain is initialized (`ignite c serve`)
2. write the consumer genesis (`ignite c init` & `ignite c serve` if 1) returns false)

As you can see these tasks are used inside multiple commands and are inside the CLI logic. Hooks are attached to the beginning or the end of the command, it's not possible to use then inside the CLI logic, nor to read value from them.

Reply:
>  i highly suggest we take time and either import all the needed code for validating consumer genesis from ics repo to ours 

Adding maintenance task like that is not something I support, but if you guys are OK for it, why not.

> or change the plugin interface so that it gives plugins ability to participate in scaffolding chains, modules and ...

If you look from the top, this is exactly the ability offered by `plugin.Execute` :)
If you want more decorations, feel free to add them!

Reply:
@Ehsan-saradar @julienrbrt this how it looks with the plugin usage: b10e62661ec7d6e0e3471a36b321e2c2add51631

Thanks to Ehsan's comment, I added some more convenient methods around `plugininternal.Execute()`.